cmd/main.go

```diff
@@ -530,7 +530,7 @@ func createStorageClient(deployTarget string, storage string, s3cfg *s3wrapper.C
 				log.Fatal("failed to create S3 client")
 			}
 		case "filesystem":
-			storageClient = s3wrapper.NewFSClient(fsWorkDir, log, versionsHandler, isoEditorFactory)
+			storageClient = s3wrapper.NewFSClient(fsWorkDir, log, versionsHandler, isoEditorFactory, metricsAPI)
```

Comment:
It's a bit unfortunate that we will have the storage init functions diverge here. Maybe as part of looking into https://github.com/openshift/assisted-service/pull/1491/files?diff=split&w=1#r612241045 we can also find a better way to handle the FS storage init.

This is not a blocker, tho!

Reply:
note that it is already diveraged, the signature of the two storage clients is different:
```
func NewS3Client(cfg *Config, logger logrus.FieldLogger, versionsHandler versions.Handler, isoEditorFactory isoeditor.Factory) *S3Client
```
vs
```
func NewFSClient(basedir string, logger logrus.FieldLogger, versionsHandler versions.Handler, isoEditorFactory isoeditor.Factory) *FSClient {
```

Other approach could be add a scheduled thread to collect metrics periodically which will not be depended on any specific storage client, but will report stats only for file-system (s3 client will not collect any info).
Pros:
- Unified API, no divergence, not adding specific logic on for one client
Cons:
- Less accuracy

Reply:
With the recent commit in this PR, the FSClient remain unchanged, it is the decorator that will use the metricsAPI